src/parser/Quads.h

```diff
@@ -28,9 +28,13 @@ struct Quads {
   // Graph triples are inside a `GRAPH ...` clause.
   std::vector<GraphBlock> graphTriples_{};
 
-  void forAllVariables(std::function<void(const Variable&)> f);
+  // Run the function for all variables in the quads. The function may be called
+  // twice for the same variable.
+  void forAllVariables(const std::function<void(const Variable&)>& f);
```

Comment:
An even better type here is `absl::FunctionRef` from `<absl/function./function_ref.h>`, see
[here](https://github.com/abseil/abseil-cpp/blob/master/absl/functional/function_ref.h), 
As it provides the needed type-erasure + indirection without copying/owning the function object.